versioned_docs/version-5.x/testing/drink.md

```diff
@@ -1,20 +1,123 @@
+
+## When to use `DRink!`?
+
+Usually, DRink! is a good choice for the development phase of your project.
+When the iteration speed is crucial, and you want to quickly test your contracts, DRink! will offer a versatile, yet highly efficient testing environment.
+However, you must not forget that it is not a replacement for the E2E tests, which should be run before the deployment to the production network, as well as in your CI/CD pipelines.
+
+DRink! also comes in handy when you need to:
+ - mock some parts of your contract suite
+ - debug the execution process
+ - launch long-running simulations, that would normally take a lot of time when relying on a real block-time
+
+## How to use DRink!?
+
+There are three ways to use DRink!:
+
+### Directly as a library
+
+This way you gain access to full DRink! power in your test suites.
+
+`drink` library is continuously published to [crates.io](https://crates.io/crates/drink), so you can use it in your project with either `cargo add drink` or by adding the following line to your `Cargo.toml`:
+```toml
+drink = { version = "0.8" }
+```
+
+Then, you can write your tests like this:
+```rust
+#[cfg(test)]
+mod tests {
+    /// This will take care of building all contract dependencies in the compilation phase and gather all contract 
+    /// bundles (metadata and the compiled code) into a single registry.
+    #[drink::contract_bundle_provider]
+    enum BundleProvider {}
+
+   /// Within `drink::test` macro, you are provided with a `session` object, which is a wrapper around the
+   /// blockchain state. You can use it to deploy contracts, call their methods, and more.
+    #[drink::test]
+    fn deploy_and_call_a_contract(mut session: Session) -> Result<(), Box<dyn Error>> {
+        let result: bool = sesison
+            .deploy_bundle_and(BundleProvider::local(), "new", &["true"], NO_SALT, NO_ENDOWMENT)?
+            .call_and("flip", NO_ARGS, NO_ENDOWMENT)?
+            .call_and("flip", NO_ARGS, NO_ENDOWMENT)?
+            .call_and("flip", NO_ARGS, NO_ENDOWMENT)?
+            .call("get", NO_ARGS, NO_ENDOWMENT)??;
+        assert_eq!(result, false);
+    }
+}
+```
+
+You can check some helpful and verbose examples [here](https://github.com/inkdevhub/drink/tree/main/examples), including the [**quick start guide**](https://github.com/inkdevhub/drink/tree/main/examples/quick-start-with-drink).
+
+### As an alternative backend to ink!'s E2E testing framework
+
+DRink! is already integrated with the ink! framework and can be used as a drop-in replacement for the standard E2E testing environment.
+Just use corresponding argument in the test macro:
+```rust
+#[ink_e2e::test(backend(runtime_only()))]
+```
+to your test function and you have just switched from E2E testcase to DRink!-based one, that doesn't use any running node in the background!
+
+For a full example check out [ink! repository](https://github.com/paritytech/ink/blob/master/integration-tests/e2e-runtime-only-backend/lib.rs).
```

Comment:
I'll update this link to point to https://github.com/paritytech/ink-examples once we've updated that repo to 5.0. It'll then contain the content of the ink! repositories `integration-tests` folder.